test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -136,8 +153,97 @@ g.test_fence_during_confirm_wal_write = function(g)
             election_timeout = box.NULL,
         }
     end)
+    server_wait_wal_is_blocked(g.server3)
+    server_unblock_wal_write(g.server3)
     g.server1:exec(function()
         box.ctl.promote()
     end)
     wait_fullmesh(g)
 end
+
+g.test_vote_during_txn_wal_write = function(g)
+    --
+    -- Make the following topology:
+    --   server1 <-> server2 <-> server3
+    --
+    g.server1:exec(function()
+        local replication = table.copy(box.cfg.replication)
+        rawset(_G, 'old_replication', table.copy(replication))
+        table.remove(replication, 3)
+        box.cfg{replication = replication}
+    end)
+    g.server3:exec(function()
+        local replication = table.copy(box.cfg.replication)
+        rawset(_G, 'old_replication', table.copy(replication))
+        table.remove(replication, 1)
+        box.cfg{replication = replication}
+    end)
+    --
+    -- Server2 gets a foreign txn going to WAL too long.
+    --
+    server_block_next_wal_write(g.server2)
+    local f = fiber.new(g.server1.exec, g.server1, function()
```

Comment:
I remember there were issues with hanging one `exec` and trying to perform another one to the same server later.
I'd rather make the exec return immediately by doing something like
```lua
g.server1:exec(function()
    require('fiber').new(....)
end)
```
But if the current version works fine for you, then never mind. Just have to be careful with that.

Reply:
AFAIK the hang can only happen if:
- The first exec killed the server;
- The first exec made the server run a busy-loop without yields;
- The first exec caused connection termination and the server rebound its server to another port;
- The server was configured to have only one fiber for serving iproto requests and it is still busy with the first request.

In this case neither of these should happen. This hack about returning fiber ids I don't like, because it forces to do `require('fiber')` on the server side every time when you want to start some async work there. It just pads the code out. I don't mind seeing it in other patches, but would want to avoid in mine.

Reply:
> In this case neither of these should happen. 

Yes, I was thinking the same, but still I remember hitting this issue in either one of mine or @grafin's patches. I don't remember which one.

Anyway, leave this version if you like it. I see that it works now, so I don't mind.